### ISSUE TITLE
Validate entry_points.txt on install, fail if necessary ?

### ISSUE THREAD

**Issue by Carreau** [takluyver/flit#96]
A bad groupname can prevent other packages from starting as pkg_resources can raise `raise ValueError("Invalid group name", group)`

**Comment by takluyver** [comment]
I get the impression you hit this? What was the example? I think it makes
sense to verify them when building the package.

**Comment by Carreau** [comment]
> I get the impression you hit this? What was the example? I think it makes
> sense to verify them when building the package.

Yes, I did a package `flitlicenses` with an entrypoint group named `[flit-licenses]` (dash is invalid) and IPython was broken on startup (because pkg_resources scan and validate even non-IPython entrypoints...). Obviously I fixed it immediately and it was my fault, but I would still have prefered to be warned at install time with failure that later. And I was lucky enough to know from where it was coming.

**Comment by takluyver** [comment]
Flit now validates the object reference part of entry points (like `flit:main`), but it doesn't do anything with the group name. Do you know what rules `pkg_resources` checks it against?

**Comment by takluyver** [comment]
Thanks @merwok - PRs to validate the group names welcome.

**Comment by earonesty** [comment]
better to use the documented pkg_resources class function `pkg_resources.EntryPoint.parse_group("group.name", [])`, instead of `MODULE`.
